### PR TITLE
ordRecord

### PR DESCRIPTION
**Pull request by matthewleon**
addresses https://github.com/justinwoo/purescript-record-extra/issues/7

### CODE REVIEW

File: src/Data/Record/Extra.purs [hunk 1]
```diff
@@ -175,6 +175,38 @@ class Applicative m <= SequenceRecord rl row row' m
...
+instance ordRecordNil :: OrdRecord Nil row where
+  ordRecordImpl _ _ _ = EQ
+
+ordRecord :: forall row rl
+   . RowToList row rl
+  => OrdRecord rl row
+  => Record row
+  -> Record row
+  -> Ordering
+ordRecord a b = ordRecordImpl (RLProxy :: RLProxy rl) a b
```
Comment (justinwoo) [inline src/Data/Record/Extra.purs]: minor nitpick, I guess this should really be called `compareRecord`?

Comment (matthewleon) [reply src/Data/Record/Extra.purs]: Good point. Will amend.

File: src/Data/Record/Extra.purs [hunk 2]
```diff
@@ -175,6 +175,38 @@ class Applicative m <= SequenceRecord rl row row' m
...
+    compareRecordImpl :: RLProxy rl -> Record row -> Record row -> Ordering
+
+instance ordRecordCons ::
+  ( IsSymbol name
+  , Ord ty
+  , RowCons name ty trash row
+  , OrdRecord tail row
+  ) => OrdRecord (Cons name ty tail) row where
+  compareRecordImpl _ a b =
+    case compare valA valB of
```
Comment (justinwoo) [inline src/Data/Record/Extra.purs:190]: another just minor thing, i think this generates code that compares the whole thing, whereas `let result = __ if ___ then ___ else ___` will generate code that does not evaluate the whole branch? might require some digging in with `spy`

Comment (matthewleon) [reply src/Data/Record/Extra.purs:190]: Here is the generated code:

```javascript
return function (b) {
    var valA = Data_Record.get(dictIsSymbol)(dictRowCons)(Data_Symbol.SProxy.value)(a);
    var valB = Data_Record.get(dictIsSymbol)(dictRowCons)(Data_Symbol.SProxy.value)(b);
    var v1 = Data_Ord.compare(dictOrd)(valA)(valB);
    if (v1 instanceof Data_Ordering.EQ) {
        return compareRecordImpl(dictOrdRecord)(Type_Row.RLProxy.value)(a)(b);
    };
    return v1;
};
```

Should be fine, no?

Comment (justinwoo) [reply src/Data/Record/Extra.purs:190]: oh, cool, seems fine then

**Pull request merged**